src/apps/advanced-search-v2/lib/suggestions.ts

```diff
@@ -0,0 +1,30 @@
+import type { Option } from "../types";
+
+// Minimal common shape from both localSuggest and complexSuggest
+// - localSuggest: { term, work: { titles: { main: string[] } } }
+// - complexSuggest: { term, work: { titles: { main: string[] } } }
+type MinimalSuggestItem = {
+  term: string;
+  work?: { titles: { main: string[] } } | null;
+};
+
+export function suggestionsToOptions(results?: MinimalSuggestItem[]): Option[] {
+  if (!results || results.length === 0) return [];
+
+  const seen = new Set<string>();
+
+  return results
+    .map((item) => {
+      const fromWork = item.work?.titles?.main?.[0];
+      const raw = (typeof fromWork === "string" && fromWork) || item.term || "";
```

Comment:
```suggestion
      const fromWork = item.work?.titles?.main?.[0] || '';
      const raw = fromWork || item.term || "";
```

Is it possible to avoid the ugly type check?